### ISSUE TITLE
field `declaration`: field required when deploying to SageMaker

### ISSUE THREAD

**Issue by aronhoff** [iterative/mlem#611]
I've been following [this tutorial](https://mlem.ai/doc/user-guide/deploying/sagemaker/#configuring-and-running-deployment) for deploying a model on SageMaker.

When I get to the step
```sh
mlem deployment run -l sagemaker_deployment -m model
```
it gives me the following error
```
⏳️ Loading model from model.mlem
⏳️ Loading deployment from sagemaker_deployment.mlem
🔗 Loading link to mlem_sagemaker.mlem
❌ Error:
❌ field `declaration`: field required
```

I tried tracking the issue down and I believe the exception is thrown on [this line](https://github.com/iterative/mlem/blob/1a1d2031643c56d24473f77c9b6a015a9d43d4a5/mlem/contrib/sagemaker/meta.py#L330) of `SagemakerDeployment.deploy`, where `SagemakerDeployState()` seems to not populate the non-optional `DeployState.declaration`.

**Comment by aguschin** [comment]
Hi @aronhoff! I'll check it out. Thanks for reporting!

**Comment by aguschin** [comment]
Hi @aronhoff! Here's the fix https://github.com/iterative/mlem/pull/615

can you try it out? How to install
```
git clone https://github.com/iterative/mlem
cd mlem
git checkout fix/sagemaker
pip install -e ".[sagemaker]"  # if you need some other extras, you can add here
```
you can do that in venv if you don't want to mess up current MLEM installation

**Comment by aronhoff** [comment]
That seems to have gotten us past this error.

The tutorial doesn't mention the `--profile` flag for the `mlem declare env sagemaker` command, so it was giving an error there.

Once I got past that, it managed to build an image and try deploy an instance. But the instance crashed because it couldn't find the docker image. I could confirm that ECR had no images. I had to manually `docker push` the image created by mlem. Then a rerun of `mlem deployment run` worked. I also had to fidget a bunch with permissions (giving the mlem user/role S3 access) and ensuring that docker was logged in to ECR, but I don't know the minimal required steps for these.

After all this, an endpoint showed up in my SageMaker, but it kept crashing with 
```
exec /bin/bash: exec format error
```
I am running mlem from an M1 Mac, so it probably built an arm64 image, rather than an amd64. I can't find a `--platform` flag to fix this though, so I'm stuck here at the moment.

That's a bunch of scope creep though showing a wider issue, strictly speaking, the MR did fix the bug I originally reported.

**Comment by aguschin** [comment]
Ok, this was partially addressed in #615, but the later error wasn't addressed there yet.

**Comment by mike0sv** [comment]
Hi [aronhoff](https://github.com/aronhoff)! PR above should fix platform issue, can you try it out? Be sure to remove base arm64 images, otherwise you'll get something like 
```
image with reference sha256:... was found but does not match the specified platform ...
```
